apps/chat/src/components/Common/ApplicationWizard/DeployableView/SourceFilesEditor.tsx

```diff
@@ -0,0 +1,97 @@
+import { IconFile, IconFolder } from '@tabler/icons-react';
+import { FC, memo, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import { getFileRootId } from '@/src/utils/app/id';
+
+import { Translation } from '@/src/types/translation';
+
+import { FilesActions, FilesSelectors } from '@/src/store/files/files.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+
+import { SelectFolderModal } from '@/src/components/Files/SelectFolderModal';
+
+interface SourceFilesEditorProps {
+  value?: string;
+  onChange?: (v: string) => void;
+}
+
+const _SourceFilesEditor: FC<SourceFilesEditorProps> = ({
+  value,
+  onChange,
+}) => {
+  const { t } = useTranslation(Translation.Settings);
+
+  const dispatch = useAppDispatch();
+
+  const files = useAppSelector(FilesSelectors.selectFiles);
+
+  const [isFolderModalOpen, setIsFolderModalOpen] = useState(false);
+
+  const folderFiles = useMemo(() => {
+    if (value) {
+      return files.filter((file) => file.id.startsWith(value));
+    }
+    return [];
+  }, [files, value]);
+
+  const handleToggleFileManager = useCallback(() => {
+    setIsFolderModalOpen((p) => !p);
+  }, [setIsFolderModalOpen]);
+
+  const handleCloseFileManager = useCallback(
+    (folder?: string) => {
+      if (folder) {
+        onChange?.(folder);
+      }
+      setIsFolderModalOpen(false);
+    },
+    [onChange],
+  );
+
+  useEffect(() => {
+    if (value) {
+      dispatch(FilesActions.getFilesWithFolders({ id: value }));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I believe this `useEffect` should depends on `value`

Reply:
This needed only when we open edit modal, if to call this useEffect on every value change we will get extra file requests because file manager also calls this action on folder select